### DISCUSSION TITLE
Terraform manager: Ignore Templating in `required_version` depType

### DISCUSSION THREAD

**Discussion by norman-zon** [renovatebot/renovate#21353] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

We use Terragrunt to have a centralised template for required provider versions with templating.

`_provider_versions.hcl`:

```

terraform {
  required_version = ">= 1.3.2"
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "4.48.0"
    }
%{ if providers.provider_github }
    github = {
      source  = "integrations/github"
      version = "5.7.0"
    }
%{ endif }
%{ if providers.provider_kubernetes }
    kubernetes = {
      source = "hashicorp/kubernetes"
      version = "2.15.0"
    }
%{ endif }
   }
}
```

I added a `file_match` in my renovate config:

```
  "terraform" :{
      "fileMatch": ["_provider_versions.hcl$"]
    },
```

But the terraform manager does not pick up the `required_versions`. My guess is the templating does not fit the expected schema.

I would like to request the ability to handle (by ignoring) templating by the terraform manager.

Thanks in advance!

**Comment by rarkins** [comment]
The manager uses the hcl parser, and if the file is not valid hcl, it doesn't get parsed. Would that explain it?

**Reply by norman-zon** [reply]
This explains ist. I'll try to find another way to handle the situation.
